metadataproxy/roles.py

```diff
@@ -170,6 +171,29 @@ def find_container(ip):
                     log.debug(msg.format(_id, ip))
                     CONTAINER_MAPPING[ip] = _id
                     return c
+        # Try to find the container over the mesos state api and use the labels attached to it
+        # as a replacement for docker env and labels
+        if app.config['MESOS_STATE_LOOKUP']:
+            try:
+                state = json.loads(urllib2.urlopen(app.config['MESOS_STATE_URL']).read())
```

Comment:
Can you break this our into a separate function. Also we have requests as a (diamond) dependency, can you use that rather than directly using urllib2? We can pull requests into the requirements.txt file directly, so it's clear it's a true dependency.

Reply:
Also, you probably want to use some type of caching for data returned from this url, likely with a some kind of timestamp for a TTL. Also, you may want to fetch this data outside of the for loop, or you'll hit it numerous time, every time a container does a metadata call, which can be really frequent.